src/components/shared/TinderCard/index.tsx

```diff
@@ -0,0 +1,211 @@
+import {
+  Animated,
+  Dimensions,
+  LayoutAnimation,
+  PanResponder,
+  Platform,
+  Text,
+  UIManager,
+} from 'react-native';
+import React, { useCallback, useMemo, useState } from 'react';
+
+import styled from 'styled-components/native';
+
+const SCREEN_WIDTH = Dimensions.get('screen').width;
+const SWIPE_THRESHOLD = 0.25 * SCREEN_WIDTH;
+const SWIPE_OUT_DURATION = 250;
+
+interface DataProps {
+  [key: string]: any;
+}
+
+interface Props {
+  testID?: string;
+  onSwipeRight?: (item: DataProps) => void;
+  onSwipeLeft?: (item: DataProps) => void;
+  onCancel?: () => void;
+  data: Array<DataProps>;
```

Comment:
Try not to save the data internally. This should be handled at the outer side instead.
Also, try not to use `Array` for typing if `T` is a simple type ([referenece](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#caveats)).

`DataProps[]` is easier to read.